src/main/java/me/xdark/recaf/runtime/util/VMUtil.java

```diff
@@ -0,0 +1,57 @@
+package me.xdark.recaf.runtime.util;
+
+import sun.misc.Unsafe;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+
+/**
+ * Utility class to provide access to trusted lookup and Unsafe
+ *
+ * @author xDark
+ */
+public final class VMUtil {
+
+    private static final Unsafe UNSAFE;
+    private static final MethodHandles.Lookup LOOKUP;
+    private static final int VM_VERSION;
+
+    private VMUtil() {
+    }
+
+    /**
+     * @return {@link Unsafe}
+     */
+    public static Unsafe unsafe() {
+        return UNSAFE;
+    }
+
+    /**
+     * @return {@link MethodHandles.Lookup}
+     */
+    public static MethodHandles.Lookup lookup() {
+        return LOOKUP;
+    }
+
+    /**
+     * @return version of JVM
+     */
+    public static int vmVersion() {
+        return VM_VERSION;
+    }
+
+    static {
+        try {
+            VM_VERSION = (int) (Float.parseFloat(System.getProperty("java.class.version")) - 44);
+            Field field = Unsafe.class.getDeclaredField("theUnsafe");
```

Comment:
I've not tested it but wouldn't this fail on Java 9+ because there isn't a `sun.misc.Unsafe` anymore?

Reply:
Nope, but you are right, some methods were removed, see `Java9ClassDefiner`

Reply:
Oh... I used to think that `sun.misc.Unsafe` was removed in Java 9, but apparently not...:sweat_smile:
They only moved it to the `jdk.unsupported` module and made it an wrapper for `jdk.internal.misc.Unsafe`.

Reply:
Oh... I used to think that `sun.misc.Unsafe` was removed in Java 9, but apparently not...:sweat_smile:
They only moved it to the `jdk.unsupported` module and made it an wrapper for `jdk.internal.misc.Unsafe`.